src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.Aot.cs

```diff
@@ -1279,4 +1282,15 @@ protected void EmitCleanUpNativeData(ILCodeStream codeStream)
             codeStream.Emit(ILOpcode.callvirt, emitter.NewToken(cleanupNativeDataMethod));
         }
     }
+
+    class BlittableValueClassByRefReturn : Marshaller
+    {
+        protected override void SetupArgumentsForReturnValueMarshalling()
+        {
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+
+            _managedHome = new Home(emitter.NewLocal(ManagedParameterType), ManagedParameterType, isByRef: false);
```

Comment:
We just need to create locals of the right type. The default implementation works fine otherwise.